src/api/base/index.ts

```diff
@@ -74,19 +101,33 @@ export default class BaseApi {
 
     const chainId = networkId || (await this.context.chainId)
 
-    let response
-    if (env === undefined) {
+    // `env` is not set, meaning we should try both envs
+    if (!env) {
+      let prodResponse
+      let barnResponse
       try {
-        response = await fetchFn(url, method, `${prodUri[chainId]}/${this.API_VERSION}`, data, requestOptions)
+        // Prod goes first
+        prodResponse = await fetchFn(url, method, `${prodUri[chainId]}/${this.API_VERSION}`, data, requestOptions)
       } catch (error) {
-        response = await fetchFn(url, method, `${barnUri[chainId]}/${this.API_VERSION}`, data, requestOptions)
+        barnResponse = await fetchFn(url, method, `${barnUri[chainId]}/${this.API_VERSION}`, data, requestOptions)
+      }
+      // If `barnResponse` is set it means `prod` fetching threw and `barn` succeeded
+      // Return that, then
+      if (barnResponse) {
```

Comment:
I get it, but i have to say is a bit convoluted for quite simple logic.
Just the fact two parts of this snipped do the actual call to barn.

One simple way is to do `fetchMultipleEnvs` receive the array of envs to try. So you can call:
[prod, barn], [barn, prod], [prod], [barn]

The method can be recursive. The base case is when there's only one item in the array.

Reply:
Addressed on https://github.com/cowprotocol/cow-sdk/pull/73